src/taxonomy/taxonService.ts

```diff
@@ -0,0 +1,56 @@
+import { User } from 'src/auth'
+import { JobStatus } from 'src/job'
+import { Taxon } from 'src/taxonomy/taxon'
+
+export interface TaxonService {
+  // ==== READ
+  count(options: { surveyId: number; taxonomyUuid: string; draft?: boolean }): Promise<number>
+
+  get(options: { surveyId: string; taxonomyUuid: string; draft: boolean }): Promise<Taxon>
+
+  getMany(options: { surveyId: string; taxonomyUuid: string; draft: boolean }): Promise<Array<Taxon>>
+
+  getByVernacularName(options: {
+    surveyId: number
+    taxonomyUuid: string
+    filterValue: string
+    draft: boolean
+    includeUnlUnk: boolean
+  }): Promise<Array<Taxon>>
+
+  getByCode(options: {
+    surveyId: number
+    taxonomyUuid: string
+    filterValue: string
+    draft: boolean
+    includeUnlUnk: boolean
+  }): Promise<Array<Taxon>>
+
+  getByScientificName(options: {
+    surveyId: number
+    taxonomyUuid: string
+    filterValue: string
+    draft: boolean
+    includeUnlUnk: boolean
+  }): Promise<Array<Taxon>>
+
+  getByVernacularNames(options: {
+    surveyId: number
+    taxonomyUuid: string
+    draft: boolean
+    limit: number
+    offset: number
+  }): Promise<Array<Taxon>>
```

Comment:
they should all be merged into getMany.
pass an option searchField?:'code'|'scientificName'|'vernacularNames', searchValue?:string.
plus, draft, limit, offset, includeUnlUnk are optional